e2e/upgrade-testing-rbd.go

```diff
@@ -0,0 +1,148 @@
+package e2e
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo" // nolint
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var (
+	err error
+
+	// cwd stores the initial working directory.
+	cwd string
+
+	// subvolumegroup contains the subvolume group name
+	// used for e2e testing.
+	subvolumegroup string
+
+	// vaultDeploy validates if vault needs to be deployed in the
+	// upgrade testing csi release.
+	vaultDeploy bool
+
+	// skipPSP makes sure that PSP are created
+	// only if supported by that release, otherwise skip.
+	skipPSP bool
+)
+
+var _ = Describe("RBD Upgrade Testing", func() {
+	f := framework.NewDefaultFramework("upgrade-test-rbd")
+	var c clientset.Interface
+	// deploy rbd CSI
+	BeforeEach(func() {
+		if !upgradeTesting || !testRBD {
+			Skip("Skipping RBD Upgrade Testing")
+		}
+		c = f.ClientSet
+		if cephCSINamespace != defaultNs {
+			err = createNamespace(c, cephCSINamespace)
+			if err != nil {
+				Fail(err.Error())
+			}
+		}
+		createNodeLabel(f, nodeRegionLabel, regionValue)
+		createNodeLabel(f, nodeZoneLabel, zoneValue)
+
+		// fetch current working directory to switch back
+		// when we are done upgrading.
+		cwd, err = os.Getwd()
+		if err != nil {
+			Fail(err.Error())
+		}
+
+		vaultDeploy, err = checkIfVaultDeploymentRequired(upgradeVersion)
```

Comment:
Even we can remove this one as the minimum version is 2.1.2. let's not do much work

Reply:
At the current stage, the job can even check for backward compatibility with `v2.0.0`. If you guys suggest so, I can remove all the extra checks, and make a minimum version for upgrade testing to be `v2.1.2`. What do you guys suggest?